src/AzureAuth/CommandMain.cs

```diff
@@ -217,6 +222,46 @@ public static string PrefixedPromptHint(string promptHint)
             return $"{PromptHintPrefix}: {promptHint}";
         }
 
+        /// <summary>
+        /// Generates event data from the AuthFlowResult.
+        /// </summary>
+        /// <param name="result">The AuthFlowResult.</param>
+        /// <returns>The event data.</returns>
+        public EventData GenerateEventData(AuthFlowResult result)
+        {
+            if (result == null)
+            {
+                return null;
+            }
+
+            var eventData = new EventData();
+            eventData.Add("auth_mode", result.AuthFlowName);
+            eventData.Add("success", result.Success);
+
+            var correlationIDs = new List<string>();
+
+            if (result.Errors.Any())
+            {
+                var error_messages = JsonSerializer.Serialize(result.Errors.Select(error => ExceptionsExtensions.ToFormattedString(error)));

```

Comment:
Is this readable? Or should I break it into two lines?

Reply:
This is fine, but I think this is the limit (for me anyway). 😅